test/e2e/storage/testsuites/snapshotsecrets.go

```diff
@@ -0,0 +1,253 @@
+					framework.ExpectNoError(sr.CleanupResource())
+				})
+				// create a secret
+				secret = &v1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: f.Namespace.Name,
+						Name:      "snapshot-secret",
+					},
+					Data: map[string][]byte{
+						"secret-data": []byte("secret-value-1"),
+					},
+				}
+
+				if secret, err = cs.CoreV1().Secrets(f.Namespace.Name).Create(context.TODO(), secret, metav1.CreateOptions{}); err != nil {
+					framework.Failf("unable to create test secret %s: %v", secret.Name, err)
+				}
+
+				// Create snapshot using secret
+				parameters := map[string]string{
+					"csi.storage.k8s.io/snapshotter-secret-name":      "snapshot-secret",
+					"csi.storage.k8s.io/snapshotter-secret-namespace": f.Namespace.Name,
+				}
+				sr = CreateSnapshotResource(sDriver, config, pattern, pvc.GetName(), pvc.GetNamespace(), parameters)
+				vs = sr.Vs
+			})
+			ginkgo.It("Use secrets field, create/list and delete snapshots", func() {
+
+				ginkgo.By("list the snapshot")
+				// TODO: Check if list snapshot is supported. Log and continue if not supported.
+				ListSnapshotsWithSecrets(dc, vs.GetNamespace(), vs.GetName(), framework.Poll, framework.SnapshotDeleteTimeout)
+				// Snapshots should be allowed to be deleted even if secrets are deleted.
+				// if secret != nil {
+				// 	framework.Logf("Deleting the secret %q...", secret.Name)
+				// 	err := cs.CoreV1().Secrets(f.Namespace.Name).Delete(context.TODO(), secret.Name, metav1.DeleteOptions{})
+				// 	if err != nil {
+				// 		framework.Logf("Delete secret failed: %v", err)
+				// 	}
+				// }
+				ginkgo.By("Delete the snapshot")
+				DeleteSnapshotWithSecrets(dc, vs.GetNamespace(), vs.GetName(), framework.Poll, framework.SnapshotDeleteTimeout)
+				ginkgo.By("checking if the snapshot is deleted")
+
+			})
+		})
+	})
+}
+
+// ListSnapshotsWithSecrets lists a given volumesnapshot it uses secret provided within storageclass
+func ListSnapshotsWithSecrets(dc dynamic.Interface, ns string, snapshotName string, poll, timeout time.Duration) error {
+	ginkgo.By("listing the snapshot")
+	// TODO: list snapshots using secrets
+	vs, err := dc.Resource(SnapshotGVR).Namespace(ns).Get(context.TODO(), snapshotName, metav1.GetOptions{})
+
+	snapshotStatus := vs.Object["status"].(map[string]interface{})
+	snapshotContentName := snapshotStatus["boundVolumeSnapshotContentName"].(string)
+	framework.Logf("received snapshotStatus %v", snapshotStatus)
+	framework.Logf("snapshotContentName %s", snapshotContentName)
+	framework.ExpectNoError(err)
+	return nil
+}
+
+// DeleteSnapshotWithSecrets deletes a VolumeSnapshot and waits for it to be deleted or until timeout occurs, whichever comes first
+func DeleteSnapshotWithSecrets(dc dynamic.Interface, ns string, snapshotName string, poll, timeout time.Duration) error {
+	var err error
+	ginkgo.By("deleting the snapshot")
+	err = dc.Resource(SnapshotGVR).Namespace(ns).Delete(context.TODO(), snapshotName, metav1.DeleteOptions{})
```

Comment:
Before deleting snapshot, verify deletion secrets is in annotation for the content.

Reply:
done

Reply:
I meant check delete snapshot secret annotation on the content.

I think this Delete call should be removed because CleanupResource will delete snapshot resources.